### DISCUSSION TITLE
Doubt about TSMP58000 alternatives

### DISCUSSION THREAD

**Discussion by morganflint** [bengtmartensson/IrScrutinizer#483] [Hardware issues] [answered]
First of all, thanks for this tool!

First, a little background for my question. I've been using IrScrutinizer along with [Arduino Nano ](http://www.harctoolbox.org/arduino_nano.html) to try to reverse engineer the protocol of a non-common AC unit, and have been able to extract the information using TSOP4838 as a demodulating receiver.

I find a slight discrepancy between the frequency reported by IrScutinizer (38k) and the one I can measure with an oscilloscope (42k), which I understand may be due to the TSOP4838 receiver's intended frequency (38kHz). It doesn't seem to affect the quality of the decoding, but it made me curious to explore the possibility to use a non-demodulating receiver.

As I don't have any on hand, I have been looking where to buy TSMP58000 or it's alternatives listed on [http://www.harctoolbox.org/arduino_nano_part2.html#Alternative+Components](http://www.harctoolbox.org/arduino_nano_part2.html#Alternative+Components), and here is where my doubt arises: the first alternatives listed are TSMP58138, TSMP1138, and TSMP4138 but, according to their datasheets (e.g., [this one](https://www.vishay.com/docs/82486/tsmp58138.pdf)), they are also demodulating receivers, with a block diagram quite similar to TSOP 4838 or TSOP34438, and quite different to ttaht of [TSMP58000](https://www.vishay.com/docs/82485/tsmp58000.pdf)

Does this mean that a demodulating receiver can also be connected to D8 and be used for signal analysis?

I've tried connecting the same TSOP4838 to D8 instead of D5, unchecking "Use receive for capture in the Girs Client tab and, although the signal plot isn't very clean, the results of the scrutinized signal (after manually correcting the zero-length replacement)  are quite similar to those obtained on D5 and "Use receive..." checked.

Maybe the difference between TSMP58138 and other demodulating receivers is the wider frequency response (AC coupled response from 30 kHz to 60 kHz, states the datasheet), but I'd like to know your view about this and the preference between TSMP58138, TSMP1138, and TSMP4138 and the other alternatives listed. Thanks a lot!!

**Comment by bengtmartensson** [comment, marked as answer]
First and foremost, a demodulating receiver and a non-demodulating receiver have completely different usage. A non-demodulating receiver is for "learning", and can provide high-precision measurements on the sent signal, but have a short range, and is sensitive to disturbanes. A demodulating sensor is for the use case of deployment reception of known IR signals; it has a very long range, very good suppression of unwanted signals, but (since it kills the modulation) does not provide very exact information on the signal's timing.

Connecting a demodulating receiver to D8 is not a good idea, and not supported.

All I know about the different non-demodulating sensors is written in the article you linked. The only thing to add is that the "small" ones (QSE15* etc) has a very short range (centimeters) while TSMP58000 etc has a range of around 2 m.

**Reply by morganflint** [reply]
Ok, thank you very much for your quick response!

I understand what you say, but I still have the doubt about **why TSMP58138, TSMP1138, and TSMP4138** (demodulating receivers according to their datasheets, if I'm not interpreting them badly) ,**are listed in the article as possible alternatives to TSMP58000** (non-demodulating).

This is the paragraph from http://www.harctoolbox.org/arduino_nano_part2.html#Alternative+Components I'm talking about:
"_(Some) alternatives to the TSMP58000 are TSMP58138, TSMP1138, and TSMP4138 (the latter is found to the right on the IrScrutinizer splash screen). They all have similar properties, very good amplification, but ends at around 60kHz modulation frequency._"

The first picture below, from the [datasheet of TSMP58000](https://www.vishay.com/docs/82485/tsmp58000.pdf), is its block diagram, completely different from the second picture, from [TSMP58138's datasheet](https://www.vishay.com/docs/82486/tsmp58138.pdf) which, in turn, is the same as the block diagram of [TSOP34438](http://www.vishay.com/docs/82489/tsop322.pdf) (3rd picture). 

So, if TSMP58138, TSMP1138, and TSMP4138 are modulating receivers: Why are they listed as alternatives to TSMP58000? There must be an error, either in this paragraph of the article or in my interpretation of the datasheets.

Again, thank you very much for your attention and your work!

**TSMP58000 block diagram:**
![image](https://user-images.githubusercontent.com/10438424/225876047-e5cbd3e7-0356-4122-aeff-97134bc1c0b8.png)

**TSMP58138 block diagram:**
![image](https://user-images.githubusercontent.com/10438424/225876467-31cdb06d-1210-4fe9-bde8-fb2cfe777a87.png)

**TSOP34438 block diagram:**
![image](https://user-images.githubusercontent.com/10438424/225876316-eb02e653-7609-453a-a110-0ab37a2855dd.png)

**Comment by bengtmartensson** [comment]
> why TSMP58138, TSMP1138, and TSMP4138 (demodulating receivers according to their datasheets,

No, they are not. Read more carefully.

I have already told you all I know. I have not made any test which is the "best". If you do that, I would be interested to know the outcome.

**Reply by morganflint** [reply]
Ok, thanks!

You're completely right! Looking more carefully at the block diagrams, the one for TSM58138 says "comparator" in the last block before the output transistor, while the one for TSOP34438 says "demodulator". It was the general appearance that fooled me, and also fooled the "38" at the end of the component's name that uses to be the frequency in demodulator receivers.

Sorry for the inconvenience, I'm just ordering TSM58138 and will report here about the results.
